### PR TITLE
Feature/MOB-4472 add to calendar component

### PR DESCRIPTION
**Pull request by AleksanderWedrychowskiKarhoo**
## JIRA
[MOB-4472](https://karhoo.atlassian.net/browse/MOB-4472)

## DESCRIPTION
[Create a description of what you have changed in this branch]

## Types of changes

What types of changes does your code introduce to the project?
_Put an `x` in the boxes that apply_

- [ ] Bugfix 
- [x] New feature 
- [ ] Tech Debt
- [ ] Breaking change 
- [ ] Documentation Update
- [ ] Other: *define what type of change it is*

## CHECKLIST

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. This is simply a reminder of what we are going to look for before merging code._
Please review the guidelines for contributing to this repository.

- [ ] All work relating to the ticket is complete (Code complete/Acceptance Criteria met)
- [ ] Lint and unit tests pass locally (if appropriate)
- [ ] Added tests that prove the fix is effective or that feature works (if appropriate)
- [ ] Documentation (if appropriate)

## SCREENSHOTS
[Add screenshots if any for changes you have made to the App/UI]

### CODE REVIEW

**Comment by AleksanderWedrychowskiKarhoo** [comment]
Simulator Screen Shot - iPhone 14 - 2022-11-27 at 17 51 40
Simulator Screen Shot - iPhone 14 - 2022-11-27 at 17 59 32

**Comment by AleksanderWedrychowskiKarhoo** [comment]
Testing:
paste the code as a first item in QuoteList header views array

```
UIHostingController(
                rootView: KarhooAddToCalendarView(
                    viewModel: .init(
                        state: .add,
                        onAddAction: { viewModel in
                            let trip = TripInfo(
                                origin: TripLocationDetails(displayAddress: "Sample origin display address"),
                                destination: TripLocationDetails(displayAddress: "Sample destination display address"),
                                dateScheduled: Date().addingTimeInterval(60*60*24),
                                quote: TripQuote(qtaHighMinutes: 60),
                                fleetInfo: FleetInfo(name: "Sample-Fleet-Name"),
                                flightNumber: "Sample-Flight-Number",
                                trainNumber: "Sample-Train-Number"
                            )
                            self.calendarWorker.addToCalendar(trip) { addedSuccessfully in
                                if addedSuccessfully {
                                    viewModel.state = .added
                                }
                            }
                        })
                )
            ).then {
                $0.loadViewIfNeeded()
                $0.view.translatesAutoresizingMaskIntoConstraints = false
                $0.view.backgroundColor = .clear
            }.view!,
```

File: KarhooUISDK/CommonUI/AddToCalendar/AddToCalendarView.swift [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+
+/// View Presenting add-to-calendar option. It does not have any actual logic implemented. To add event to the calendar, please handle `viewModel.onAddAction` and use `AddToCalendarWorker`.
+/// If succeeded, set `viewModel.state` to `.added`, to change design from button, to success information label.
+struct KarhooAddToCalendarView: View {
+
+    // MARK: - Nested types
+
+    private enum Constants {
+        static let height: CGFloat = 28
+        static let calendarIconSide: CGFloat = 20
```
Comment (BartlomiejSopalaKarhoo) [inline KarhooUISDK/CommonUI/AddToCalendar/AddToCalendarView.swift:19]: Can we use something from UIConstans: Dimension.Button and Dimension.Icon?

Comment (AleksanderWedrychowskiKarhoo) [reply KarhooUISDK/CommonUI/AddToCalendar/AddToCalendarView.swift:19]: Not really. I tried using commonly used values, and it was not looking good ;/

**Review by dianakarh — APPROVED** [review]

**Pull request merged**